pkg/kwok/controllers/node_controller.go

```diff
@@ -615,3 +632,21 @@ func (c *NodeController) funcNodeName() string {
 func (c *NodeController) funcNodePort() int {
 	return c.nodePort
 }
+
+// backoffDelay calculates the backoff delay of a Stage job based on its failed times
+func (c *NodeController) backoffDelay(failedTimes int) time.Duration {
+	if failedTimes <= 0 {
+		return 0
+	}
+
+	delay := float64(c.backoff.Duration)
+
+	for i := 0; i < failedTimes; i++ {
+		delay *= c.backoff.Factor
+		if delay >= float64(c.backoff.Cap) {
+			delay = float64(c.backoff.Cap)
+			break
+		}
+	}
```

Comment:
```suggestion
	delay := math.Min(float64(c.backoff.Duration) * math.Pow(c.backoff.Factor, float64(failedTimes)), c.backoff.Cap)
```

Reply:
I use a raw for loop here since  `failedTimes`  may grow very large and cause `math.Pow` to overflow?

Reply:
`math.Pow` too large returns +Inf, which can also be used for `math.Min`, 

``` golang
math.Pow(2, 10000) # +Inf
math.Min(math.Pow(2, 10000), 100) # 100
```